SalesforceCommerceCloud/pwa-kit
# [V3][Hooks Integration] Fix Create Basket Race Condition

# Description

Currently we use a hook called `useCurrentBasket` throughout the app. This hook does double duty and creates a new basket if the customer doesn't currently have an assigned to them. Its used to create a new basket after checkout, but the following happen:

- An order is created
- The current basket is removed and invalidates the customer baskets.
- All active instances of the `useCurrentBasket` hook are recalculated getting the current baskets
- Once those request succeed (4 of them) they invoke the `onSuccess` callback, which all but one fail.

The solution is the simplify the `useCurrentBasket` hook and remove the basket creation logic. By moving this into the `<App/>` component where it will only ever create a basket in one place we know we don't have a race condition because there is only one instance of the App component.

The downside is that there is a little duplication of code, but I don't think it outweighs that this is probably a better place to do basket creation. 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Move basket creation to the app component.

# How to Test-Drive This PR

- Run through a guest checkout and look at your network tab to see that there are no failed calls to the baskets endpoint.
- Confirm that you can add something to your (new) basket after doing a checkout. (Verify the logic that will create a new basket)

# Checklists




## General
